sdk/spring/spring-cloud-azure-service/src/main/java/com/azure/spring/cloud/service/implementation/identity/api/AuthProperty.java

```diff
@@ -0,0 +1,126 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.spring.cloud.service.implementation.identity.api;
+
+import com.azure.core.util.Configuration;
+import com.azure.identity.AzureAuthorityHosts;
+
+import java.util.Map;
+import java.util.Properties;
+
+public enum AuthProperty {
+
+    CLIENT_ID("azure.clientId",
+        "Client ID to use when performing service principal authentication with Azure.",
+        false),
+    CLIENT_SECRET("azure.clientSecret",
+        "Client secret to use when performing service principal authentication with Azure.",
+        false),
+    CLIENT_CERTIFICATE_PATH("azure.clientCertificatePath",
+        "Path of a PEM certificate file to use when performing service principal authentication with Azure.",
+        false),
+    CLIENT_CERTIFICATE_PASSWORD("azure.clientCertificatePassword",
+        "Password of the certificate file.",
+        false),
+    USERNAME("azure.username",
+        "Username to use when performing username/password authentication with Azure.",
+        false),
+    PASSWORD("azure.password",
+        "Password to use when performing username/password authentication with Azure.",
+        false),
+    MANAGED_IDENTITY_ENABLED("azure.managedIdentityEnabled",
+        "Whether to enable managed identity to authenticate with Azure.",
+        false),
+    AUTHORITY_HOST("azure.authorityHost",
+        AzureAuthorityHosts.AZURE_PUBLIC_CLOUD,
+        "The well known authority hosts for the Azure Public Cloud and sovereign clouds.",
+        true),
+
+    TENANT_ID("azure.tenantId",
+        "Tenant ID for Azure resources.",
+        true),
+    CLAIMS("azure.claims",
+        "Claims for Azure resources.",
+        false),
+    SCOPES("azure.scopes",
+        "Scopes for Azure resources.",
+        false),
+
+    GET_TOKEN_TIMEOUT("azure.accessTokenTimeoutInSeconds",
+        "Max time to get AccessToken.",
+        false),
+    TOKEN_CREDENTIAL_PROVIDER_CLASS_NAME("azure.tokenCredentialProviderClassName",
+        "Reference to class which implements TokenCredentialProvider.",
+        false),
+
+    TOKEN_CREDENTIAL_BEAN_NAME("azure.tokenCredentialBeanName",
+        "The given bean name of a TokenCredential bean in the Spring context.",
+        false),
```

Comment:
Do you think we should set the default value of TOKEN_CREDENTIAL_BEAN_NAME here?